Content.Shared/Polymorph/Components/ChameleonDisguisedComponent.cs

```diff
@@ -0,0 +1,31 @@
+using Content.Shared.Polymorph.Systems;
+using Content.Shared.FixedPoint;
+using Robust.Shared.GameStates;
+
+namespace Content.Shared.Polymorph.Components;
+
+/// <summary>
+/// Added to a player when they use a chameleon projector.
+/// Handles making them invisible and revealing when damaged enough or switching hands.
+/// </summary>
+[RegisterComponent, NetworkedComponent, Access(typeof(SharedChameleonProjectorSystem))]
+public sealed partial class ChameleonDisguisedComponent : Component
+{
+    /// <summary>
+    /// How much damage can be taken before revealing automatically.
+    /// </summary>
+    [DataField]
+    public FixedPoint2 Integrity = FixedPoint2.Zero;
```

Comment:
This shouldnt be necessary and it just leads to jank in the code, either make the disguise inmortal but relay all damage or give it X amount of health in the prototype itself (should be like 50 health and make the projector have a cooldown of like a minute when it dies if it doesnt have one already).

Reply:
wide swings damage the user so having the disguise only vanish if you left click would be weird

Reply:
Thats fair, there's still 2 other options i mentioned tho